src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java

```diff
@@ -108,7 +108,7 @@ protected final boolean mustCheckName(DetailAST ast) {
 
         if (isStatic && isFinal && shouldCheckInScope(modifiersAST)
                 || ScopeUtil.isInAnnotationBlock(ast)
-                || ScopeUtil.isInInterfaceOrAnnotationBlock(ast)
+                || ScopeUtil.isInInterfaceBlock(ast)
                         && !ScopeUtil.isInCodeBlock(ast)) {
```

Comment:
`ScopeUtil.isInAnnotationBlock` was the one that was surviving, however this is just because the next check was doing `InterfaceOrAnnotation` which was duplicating the same check. It is basically the same as writting `A || (B || A)` which just equates to `A || B` which is what the code was re-written to.